pkg/api/secretgenerator/secretgenerator.go

```diff
@@ -0,0 +1,147 @@
+package secretgenerator
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/getlantern/deepcopy"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/ci-tools/pkg/util/gzip"
+)
+
+func LoadConfigFromPath(path string) (Config, error) {
+	var config Config
+	cfgBytes, err := gzip.ReadFileMaybeGZIP(path)
+	if err != nil {
+		return nil, err
+	}
+	if err := yaml.Unmarshal(cfgBytes, &config); err != nil {
+		return nil, err
+	}
+	return config, nil
+}
+
+type Config []SecretItem
+
+func (c *Config) UnmarshalJSON(data []byte) error {
+	var config []SecretItem
+	if err := yaml.Unmarshal(data, &config); err != nil {
+		return err
+	}
+
+	var err error
+
+	var newConfig Config
+	for _, si := range config {
+		i, _ := si.generateItemsFromParams()
+		newConfig = append(newConfig, i...)
+	}
+
+	*c = newConfig
+	if err != nil {
```

Comment:
Its impossible for this error to be non-nil and `if err != nil { return err }; return nil ` is equal to `return err`